include/cell.tcc

```diff
@@ -336,15 +336,28 @@ inline bool mpm::Cell<Tdim>::is_point_in_cell(
 
   // Check if the transformed coordinate is within the unit cell:
   // between 0 and 1-xi(1-i) if the element is a triangle, and between
-  // -1 and 1 if otherwise
+  // -1 and 1 if otherwise. Also, check if the transformed coordinate lies
+  // exactly on cell edge.
+  const double tolerance = std::numeric_limits<double>::epsilon();
   if (this->element_->corner_indices().size() == 3) {
```

Comment:
@kks32 should we do inline for triangular and quadrilateral elements? Too many `if` statements here